ow_plexil/src/plans/FaultHandlingPattern2.plp

```diff
@@ -8,47 +8,54 @@
 // If an antenna fault but no arm faults occurring, only action 2 occurs.
 // If an arm fault but no antenna faults occurring, only action 1 occurs.
 // If both antenna and arm faults occurring, the plan waits for resolution.
-// This plan must be terminated with an interrupt, e.g. Control-C.
+// For now, this plan must be terminated with an interrupt, e.g. Control-C.
+
+// NOTE: the Wait statements in this plan prevent deadlock between the
+// two concurrent nodes, by inducing a macro step (environment
+// processing cycle) that allows the other node to advance.
 
 #include "ow-interface.h"
 
-FaultHandlingPattern2:
+FaultHandlingPattern2: UncheckedSequence
 {
   Real NewAngle = 0;
 
   log_info ("Starting FaultHandlingPattern2 plan...");
 
   Run: Concurrence
   {
-
-  UncheckedSequence
-  {
-    RepeatCondition true;
-
-    Wait .2;
-
-    Pattern2PanAntenna:
+    Antenna: UncheckedSequence
     {
-      Start !Lookup(AntennaPanFault);
-      Exit Lookup(AntennaPanFault);
+      Repeat true;
 
-      NewAngle = (Lookup(PanDegrees) + 15) mod 360;
+      Pan: UncheckedSequence
+      {
+        Start !Lookup(AntennaPanFault);
+        Exit Lookup(AntennaPanFault);
 
-      if NewAngle > 180{
-        NewAngle = NewAngle - 360;
+        NewAngle = (Lookup(PanDegrees) + 15) mod 360;
+        if (NewAngle > 180) NewAngle = NewAngle - 360;
+        LibraryCall Pan (Degrees=NewAngle);
       }
-
-      LibraryCall Pan (Degrees=NewAngle);
+      Wait 0.1;
     }
-  }
 
-    Pattern2UnstowStow:
+    Arm: UncheckedSequence
     {
       Repeat true;
-      Start !Lookup(ArmFault);
-      Invariant !Lookup(ArmFault);
+      Exit Lookup(PowerFault);
```

Comment:
The expected behavior during a power fault is not described, and I wonder why PowerFault is the exit condition here, but not for the Pan action?

Reply:
Good catch, I have removed it.  This was a vestige from a previous version that had the plan terminate with a Power Fault (which was also checked while panning).   That feature uncovered a different bug, possibly in PLEXIL, that was not a candidate for a hotfix, so I just removed the feature.  Thus a Ctrl-C is still needed to end this plan.